app/pages/onboarding/04-connect-ledger/connect-ledger.tsx

```diff
@@ -38,57 +32,48 @@ export const ConnectLedger: React.FC = () => {
   const [ledgerLaunchVersionError, setLedgerLaunchVersionError] = useState<string | null>(null);
   useBackButton(routes.CREATE);
 
+  const { step, isLocked } = usePrepareLedger();
+
   const dispatch = useDispatch();
   const history = useHistory();
-  const { transport, step, error } = useLedger();
 
   async function handleLedger() {
     setDeviceError(null);
     setLoading(true);
-    const usbTransport = transport;
-
-    if (usbTransport === null) return;
-
-    const app = new BlockstackApp(usbTransport);
 
     try {
-      await app.getVersion();
-
-      const confirmedResponse = await app.showAddressAndPubKey(STX_DERIVATION_PATH);
-
-      if (confirmedResponse.returnCode === LedgerError.TransactionRejected) {
+      const deviceResponse = await api.ledger.requestAndConfirmStxAddress();
+      if (deviceResponse.returnCode === LedgerError.TransactionRejected) {
         setDidRejectTx(true);
         setLoading(false);
         return;
       }
 
-      if (confirmedResponse.returnCode !== LedgerError.NoErrors) {
-        setLoading(false);
-        return;
+      if (deviceResponse.returnCode === 0xffff) {
+        throw new Error(deviceResponse.errorMessage);
       }
 
-      if (confirmedResponse.address) {
+      if ('publicKey' in deviceResponse) {
         setLoading(true);
         setHasConfirmedAddress(true);
         await delay(750);
-
-        if (isMainnet() && !confirmedResponse.address.startsWith('SP')) {
+        if (isMainnet() && !deviceResponse.address.startsWith('SP')) {
```

Comment:
what about a multisig address? I feel like we could improve this check. I have some code here https://github.com/blockstack/explorer/blob/main/src/common/utils/addresses.ts that could help.